src/istio/chart/templates/gateway.yaml

```
@@ -39,4 +39,27 @@ spec:
         minProtocolVersion: {{ if $.Values.tls.supportTLSV1_2 }}TLSV1_2{{ else }}TLSV1_3{{ end }}
         {{- end }}
     {{ end }}
+    {{- if .Values.enableRootDomain }}
+    - hosts:
+        - "{{ $domain }}"
+      port:
+        name: "http-root-domain"
+        number: 80
+        protocol: HTTP
+      tls:
+        httpsRedirect: true
+    - hosts:
+        - "{{ $domain }}"
+      port:
+        name: "https-root-domain"
+        number: 443
+        protocol: HTTPS
+      tls:
+        mode: {{ .Values.rootDomainTLS.mode | default "SIMPLE" }}
+        {{- if ne (.Values.rootDomainTLS.mode | default "SIMPLE") "PASSTHROUGH" }}
+        credentialName: {{ .Values.rootDomainTLS.credentialName | default "root-domain-tls" | quote }}
+        # if supportTLSV1_2 is both defined and true, use TLSV1_2, otherwise use TLSV1_3
+        minProtocolVersion: {{ if .Values.rootDomainTLS.supportTLSV1_2 }}TLSV1_2{{ else }}TLSV1_3{{ end }}
+        {{- end }}
```

I don't have a strong feeling here but I wonder if we should fallback to the `.Values.tls` values for most of these if not set (and then fallback to hardcoded values if neither are set), at least for mode/supportTLSV1_2 (not sure it makes as much sense for credential)?

---

yeah i think that is valid and probably safer that way. i pulled the logic into a variable for easier reference.